test/functional/apps/dashboard/esql_controls/field_control.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const retry = getService('retry');
+  const kibanaServer = getService('kibanaServer');
+  const { dashboard, timePicker, common } = getPageObjects(['dashboard', 'timePicker', 'common']);
+  const testSubjects = getService('testSubjects');
+  const monacoEditor = getService('monacoEditor');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+  const browser = getService('browser');
+  const comboBox = getService('comboBox');
+  const elasticChart = getService('elasticChart');
+
+  describe('dashboard - add a field type ES|QL control', function () {
+    before(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+      await kibanaServer.importExport.load(
+        'test/functional/fixtures/kbn_archiver/dashboard/current/kibana'
+      );
+      await kibanaServer.uiSettings.replace({
+        defaultIndex: '0bf35f60-3dc9-11e8-8660-4d65aa086b3c',
+      });
+    });
+
+    after(async () => {
+      await dashboard.navigateToApp();
+      await testSubjects.click('discard-unsaved-New-Dashboard');
+    });
+
+    it('should add an ES|QL field control', async () => {
+      await dashboard.navigateToApp();
+      await dashboard.clickNewDashboard();
+      await timePicker.setDefaultDataRange();
+      await dashboard.switchToEditMode();
+      await dashboardAddPanel.clickEditorMenuButton();
+      await dashboardAddPanel.clickAddNewPanelFromUIActionLink('ES|QL');
+      await dashboard.waitForRenderComplete();
+
+      await retry.try(async () => {
+        const panelCount = await dashboard.getPanelCount();
+        expect(panelCount).to.eql(1);
+      });
+
+      await monacoEditor.waitCodeEditorReady('InlineEditingESQLEditor');
+
+      await retry.waitFor('control flyout to open', async () => {
+        await monacoEditor.setCodeEditorValue(''); // clear the default query
+        await monacoEditor.typeCodeEditorValue(
+          'FROM logstash* | STATS COUNT(*) BY ',
+          'InlineEditingESQLEditor'
+        );
+        // Wait until suggestions are loaded
+        await common.sleep(1000);
+        // Create control is the third suggestion
```

Comment:
This seems brittle. Would it be better to just test against a dashboard with an existing ESQL control?

Reply:
The editing is not as important as the actual creation as the flow starts from the editor. So I want to have these tests to test the important flow. They are stable, I have run the flaky tests multiple times. If you dont feel comfortable with owning them I can move them to my team. No prob at all